tests/client_test.py

```diff
@@ -254,5 +258,60 @@ def setUp(self):
         self.uut = SwaggerClient.from_resource_listing(self.resource_listing)
 
 
-if __name__ == '__main__':
-    unittest.main()
+class MockHttpClient(object):
+
+    def __init__(self, server_response):
+        self.response = server_response
+
+    def request(self, request, response_callback):
+        return HttpFuture(
+            mock.Mock(),
+            mock.Mock(return_value=mock.Mock(
+                json=mock.Mock(return_value=self.response),
+                status_code=200)),
+            response_callback)
+
+
+@pytest.fixture
+def swagger_client():
+    spec_dict = {
+        "swagger": "2.0",
+        "info": {
+            "version": "1.0.0",
+            "title": "Simple"
+        },
+        "paths": {
+            "/pet": {
+                "get": {
+                    "operationId": "getPet",
+                    "responses": {
+                        "200": {
+                            "description": "Get a pet",
+                            "schema": {
+                                "$ref": "#/definitions/Pet",
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "definitions": {
+            "Pet": {
+                "properties": {
+                    "id": {
+                        "type": "integer",
+                    }
+                }
+            }
+        }
+    }
+    return SwaggerClient(Spec.from_dict(spec_dict, 'http://example.com/'))
+
+
+def test_client_ignores_unexpecte_response_fields(swagger_client):
+    swagger_client.swagger_spec.http_client = MockHttpClient({
+        "id": 12345,
+        "alias": "thing",
+    })
+    response = swagger_client.pet.getPet()
+    assert response.result() == dict(id=12345)
```

Comment:
Maybe "ignored" could use some clarification here.  It is not ignored by bravado because `additionalProperties: true` is implicit when not specified. Hence, bravado is passing through the `alias` key to the client and it should be in the dict returned by `response.result()`. 

You can find a branch that verifies this behavior here: https://github.com/analogue/bravado/tree/spatel_test_unknown_fields

The test failure output:
```
=================================================================== FAILURES ===================================================================
________________________________________________ test_client_ignores_unexpecte_response_fields _________________________________________________

swagger_client = SwaggerClient(http://example.com/)

    def test_client_ignores_unexpecte_response_fields(swagger_client):
        # swagger_client.swagger_spec.http_client = MockHttpClient({
        #     "id": 12345,
        #     "alias": "thing",
        # })
        response = swagger_client.pet.getPet()
>       assert response.result() == dict(id=12345)
E       assert {'alias': 'th..., 'id': 12345} == {'id': 12345}
E         Omitting 1 identical items, use -v to show
E         Left contains more items:
E         {'alias': 'thing'}
E         Use -v to get the full diff

tests/client_test.py:330: AssertionError
```